Source/Charts/Renderers/BarChartRenderer.swift

```diff
@@ -108,6 +108,32 @@ open class BarChartRenderer: BarLineScatterCandleBubbleRenderer
         var barRect = CGRect()
         var x: Double
         var y: Double
+
+
+
+        // When drawing each bar, the renderer actually draws each bar from 0 to the required value.
+        // This drawn bar is then clipped to the visible chart rect in BarLineChartViewBase's draw(rect:) using clipDataToContent.
+        // While this works fine when calculating the bar rects for drawing, it causes the accessibilityFrames to be oversized in some cases.
+        // This offset attempts to undo that unnecessary drawing when calculating barRects
+        // This offset calculation is not necessary when bars won't be clipped (when y-axis minimum <= 0  and y-axis maximum  >= 0)
+        var heightOffset: CGFloat = 0.0
+        var originYOffset: CGFloat = 0.0
+        if let offsetView = dataProvider as? BarChartView
+        {
+            let offsetAxis = dataSet.axisDependency == .left ? offsetView.leftAxis : offsetView.rightAxis
+            if offsetAxis.axisMinimum > 0
+            {
+                heightOffset = CGFloat(offsetAxis.axisMinimum)
+            }
+            if offsetAxis.axisMaximum < 0
+            {
+                originYOffset = CGFloat(offsetAxis.axisMaximum)
+            }
+            if isInverted
```

Comment:
wow my skill is a little rusty, didn't realize this is a solution. I will test more